WEB/Src/WindowsServer/WindowsServer/Implementation/AzureMetadataRequestor.cs

```diff
@@ -20,15 +20,24 @@ internal class AzureMetadataRequestor : IAzureMetadataRequestor
         internal const int AzureImsMaxResponseBufferSize = 512;
 
         /// <summary>
-        /// Default timeout for the web requests made to obtain Azure IMS data. Internal to expose to tests.
+        /// Default timeout for the web requests made to obtain Azure IMS data.
         /// </summary>
-        internal TimeSpan AzureImsRequestTimeout = TimeSpan.FromSeconds(10);
+        private TimeSpan AzureImsRequestTimeout = TimeSpan.FromSeconds(10);
 
         private readonly HttpClient httpClient;
```

Comment:
Who owns the lifecycle management for this http client? Who is responsible for `Dispose`?

What is the thread safety guarantee? Do we allow concurrent invocations?

Reply:
this is a miss on my part. I need to implement the correct Dispose pattern.

This code is only ever called once, when the SDK initializes TelemetryModules. There are no retries.

https://github.com/microsoft/ApplicationInsights-dotnet/blob/a90eb8fa1013b5aaf484bc740b97d22a2df8141d/WEB/Src/WindowsServer/WindowsServer/AzureInstanceMetadataTelemetryModule.cs#L60-L89

https://github.com/microsoft/ApplicationInsights-dotnet/blob/a90eb8fa1013b5aaf484bc740b97d22a2df8141d/WEB/Src/WindowsServer/WindowsServer/Implementation/AzureComputeMetadataHeartbeatPropertyProvider.cs#L59-L103